src/pykeen/nn/modules.py

```diff
@@ -0,0 +1,1092 @@
+# -*- coding: utf-8 -*-
+
+"""Stateful interaction functions."""
+
+from __future__ import annotations
+
+import logging
+import math
+from abc import ABC, abstractmethod
+from typing import (
+    Any, Callable, Generic, Mapping, MutableMapping, Optional, Sequence, Tuple, Union,
+    cast,
+)
+
+import torch
+from torch import FloatTensor, nn
+
+from . import functional as pkf
+from ..typing import HeadRepresentation, RelationRepresentation, TailRepresentation
+from ..utils import CANONICAL_DIMENSIONS, convert_to_canonical_shape, ensure_tuple, upgrade_to_sequence
+
+__all__ = [
+    # Base Classes
+    'Interaction',
+    'FunctionalInteraction',
+    'TranslationalInteraction',
+    # Concrete Classes
+    'ComplExInteraction',
+    'ConvEInteraction',
+    'ConvKBInteraction',
+    'DistMultInteraction',
+    'ERMLPInteraction',
+    'ERMLPEInteraction',
+    'HolEInteraction',
+    'KG2EInteraction',
+    'NTNInteraction',
+    'ProjEInteraction',
+    'RESCALInteraction',
+    'RotatEInteraction',
+    'SimplEInteraction',
+    'StructuredEmbeddingInteraction',
+    'TransDInteraction',
+    'TransEInteraction',
+    'TransHInteraction',
+    'TransRInteraction',
+    'TuckerInteraction',
+    'UnstructuredModelInteraction',
+]
+
+logger = logging.getLogger(__name__)
+
+
+def _get_batches(z, slice_size):
+    for batch in zip(*(hh.split(slice_size, dim=1) for hh in ensure_tuple(z)[0])):
```

Comment:
Hi, I'm trying to learn the code. Why `dim=1` is fixed here? I've thought `dim` should be passed from `_forward_slicing_wrapper` for different h, r, t.

Reply:
Thanks @Yevgnen, we just patched this in #683